src/match.c

```diff
@@ -69,7 +72,7 @@ static bool match_rule_keys_match_filter(MatchRuleKeys *keys, MatchFilter *filte
         if (keys->filter.sender && !c_string_equal(keys->filter.sender, filter->sender))
                 return false;
 
-        if (keys->filter.destination && !c_string_equal(keys->filter.destination, filter->destination))
+        if (keys->filter.destination && keys->filter.destination != filter->destination)
```

Comment:
This won't quite work. :1.0 is a valid unique name. I guess we need to introduce a PEER_ID_INVALID constant (-1) to indicate that the ID is unset. Or something like that.